code/game/objects/structures/curtains.dm

```diff
@@ -6,11 +74,45 @@
 	force = 3 //just plastic
 	w_class = ITEM_SIZE_HUGE //curtains, yeap
 
-	var/obj/structure/curtain/holder = /obj/structure/curtain
+	var/curtain_kind_path = /decl/curtain_kind //path to decl containing the curtain's details
+
+/obj/item/curtain/bed
+	curtain_kind_path = /decl/curtain_kind/cloth/bed
+/obj/item/curtain/black
+	curtain_kind_path = /decl/curtain_kind/cloth/black
+/obj/item/curtain/bar
+	curtain_kind_path = /decl/curtain_kind/cloth/bar
+/obj/item/curtain/medical
+	curtain_kind_path = /decl/curtain_kind/plastic/medical
+/obj/item/curtain/privacy
+	curtain_kind_path = /decl/curtain_kind/plastic/privacy
+/obj/item/curtain/shower
+	curtain_kind_path = /decl/curtain_kind/plastic/shower
+/obj/item/curtain/shower/engineering
+	curtain_kind_path = /decl/curtain_kind/plastic/shower/engineering
+/obj/item/curtain/shower/security
+	curtain_kind_path = /decl/curtain_kind/plastic/shower/security
+/obj/item/curtain/canteen
+	curtain_kind_path = /decl/curtain_kind/plastic/canteen
+
+/obj/item/curtain/Initialize(ml, material_key, var/decl/curtain_kind/kind = null)
+	if(kind)
+		if(istype(kind))
+			curtain_kind_path = kind.type
+		else if(ispath(kind))
+			curtain_kind_path = kind
+	else
+		kind = GET_DECL(curtain_kind_path)
+	src.alpha = kind.alpha
+	src.set_color(kind.color)
```

Comment:
Alpha/colour set should be done in on_update_icon() after calling ..() otherwise material updates will potentially overwrite them.